scraper/scraper_utils.py

```diff
@@ -0,0 +1,116 @@
+
+        for item in filtered_data.keys():
+            if 'floor' in item:
+                filtered_data['floor'] = filtered_data['floor'].split(':')[1].strip('"}flor_')
+            if 'area' in item:
+                filtered_data['area'] = filtered_data['area'].split(':')[1].split(' ')[0].strip('"')
+            if 'rooms' in item:
+                filtered_data['rooms'] = filtered_data['rooms'].split(':')[1].split(' ')[0].strip('"')
+            if 'offer_type_of_building' in item:
+                filtered_data['offer_type_of_building'] = filtered_data['offer_type_of_building'].split(':')[1].strip('"}')
+            if 'market' in item:
+                filtered_data['market'] = filtered_data['market'].split(':')[1].strip('"}')
+            if 'longtitude' in item:
+                filtered_data['longtitude'] = filtered_data['longtitude'].split(':')[1]
+            if 'latitude' in item:
+                filtered_data['latitude'] = filtered_data['latitude'].split(':')[1]
+            if 'price' in item:
+                filtered_data['price'] = ''.join(filtered_data['price'].split(':')[1].split(' ')[0:2]).strip('"')
+    except Exception:
+        dump_scraped_data_to_pickle(str_request_data, str(uuid.uuid4()))
+
+    return check_if_data_contains_all_fields(filtered_data, SEARCHED_FIELDS_OX)
+
+def get_flat_otd_data(link: str) -> dict:
+    request = requests.get(link)
+    str_request_data = str(request.content).split(',')
+    filtered_data = {}
+    try:
+        for item in str_request_data:
+            if SEARCHED_FIELDS_OTD['offer_type'] in item:
+                filtered_data['offer_type'] = item.split(':')[1].strip('"}')
+            if SEARCHED_FIELDS_OTD['floor'] in item:
+                filtered_data['floor'] = item.split(':')[1].strip('[]"_flor')
+            if SEARCHED_FIELDS_OTD['area'] in item:
+                filtered_data['area'] = item.split(':')[2].strip('[]"_flor')
+            if SEARCHED_FIELDS_OTD['rooms'] in item:
+                filtered_data['rooms'] = item.split(':')[1].strip('[]"')
+            if SEARCHED_FIELDS_OTD['offer_type_of_building'] in item:
+                filtered_data['offer_type_of_building'] = item.split(':')[1].strip('[]"')
+            if SEARCHED_FIELDS_OTD['market'] in item:
+                filtered_data['market'] = item.split(':')[1].strip('[]"')
+            if SEARCHED_FIELDS_OTD['longtitude'] in item:
+                filtered_data['longtitude'] = item.split(':')[1].strip('[]"')
+            if SEARCHED_FIELDS_OTD['latitude'] in item:
+                filtered_data['latitude'] = item.split(':')[1].strip('[]"')
+            if SEARCHED_FIELDS_OTD['price'] in item:
+                filtered_data['price'] = item.split(':')[1].strip('[]"')
+    except Exception:
+        dump_scraped_data_to_pickle(str_request_data, str(uuid.uuid4()))
+    
+    return check_if_data_contains_all_fields(filtered_data, SEARCHED_FIELDS_OTD)
```

Comment:
Those `if`s are quite unreadable. It should be possible to parse json and access elements of interest by key, which would simplify code a little.

```python
text = response.content.decode('utf-8')
m = re.search(r'<script id="__NEXT_DATA__".*>(.*)</script>', text)
data = json.loads(m.groups()[0])
target = data['props']['pageProps']['ad']['target'] | data['props']['pageProps']['adTrackingData']

SEARCHED_FIELDS_OTD_TRANSLATION = {
    'price': 'ad_price',
    "offer_type": 'user_type',
    "floor": 'Floor_no',
    "area": 'Area',
    "rooms": 'Rooms_num',
    "offer_type_of_building": 'ProperType',
    "market": 'market',
    "longtitude": 'long',
    "latitude": 'lat',
    'city_name': 'city_name',
    'build_year': 'Build_year'
}

result = {}
for target_key, raw_key  in SEARCHED_FIELDS_OTD_TRANSLATION.items():
    v = target.get(raw_key)
    if isinstance(v, list):
        v = v[0]
    result[target_key] = v
```

or just:

```python
result2 = {
    'price': data['props']['pageProps']['adTrackingData']['ad_price'],
...
}
```